research/second_llm_pass/ds_gen/generate_coupon_selection_ds.py

```diff
@@ -0,0 +1,312 @@
+from os import getenv
+import sys
+from typing import List, Tuple, Dict, Optional, Callable, TypedDict
+import re
+import json
+import datasets
+import pandas as pd
+from huggingface_hub import login
+from huggingface_hub import HfApi
+
+
+# Constants
+TAG_B_COUPON = 'B-COUPON' # begin coupon tag
+TAG_I_COUPON = 'I-COUPON' # inside coupon tag
+TAG_UNKNOWN = 'UNKNOWN' # not-a-coupon tag
+
+COL_TEXT_FULL = 'content_full' # column from coupons frame with full coupon text
+COL_CONTENT_TEXT = 'text' # column from content_generic file containing text
+COL_GROUPBY = 'id' # column from both coupons and content_generic to group by values
+COL_VIEW_ID = 'view_id' # column for view id in content_generic frame
+COL_DEPTH = 'view_depth' # column with view depth from content_generic frame
+
+COL_IS_COUPON = 'is_coupon' # newly created column with flag indicating that a row is part of a coupon; internal use
```

Comment:
```suggestion
__COL_IS_COUPON = 'is_coupon' # newly created column with flag indicating that a row is part of a coupon; internal use
```
This way you can force it to be internal (as long as you don't play with name mangling this will hide the variable from the outside world.)

Reply:
done